parsec/backend/invite.py

```diff
@@ -201,8 +203,16 @@ def generate_invite_email(
             use_ssl=backend_addr.use_ssl,
         )
     )
+    http_invite_link = parsec_invite_link.replace("parsec://", "")
```

Comment:
This cooking is too fragile (for instance what if `http_invite_link` contains multiple `/` ?)

Ideally we should create a new class `BackendHTTPRedirectionAddr`, so we could do something like:
```python
invite_addr = BackendInvitationAddr.build(...)
http_redirection_addr = BackendHTTPRedirectionAddr.build(
    invite_addr=invite_addr
)
```

or we could even add a helper method in `BackendInvitationAddr` to simplify this:
```python
invite_addr = BackendInvitationAddr.build(...)
http_redirection_addr = invite_addr.to_http_redirection_addr()
```

The trouble is `BackendAddr` (the base class all addr classes inherit from) consider the scheme is always `parsec://`, so modifying this would be hazardous (and implementing a `BackendHTTPRedirectionAddr` without making it inherit from `BackendAddr` means copy/pasting a long non-trivial code).
In the end I don't think it is worth it given this redirection url is not used a lot.

All in all, I think we should only add the `BackendInvitationAddr.to_http_redirection_url()` method which would return the redirection as a `str` (note the naming: we call `addr` the object and `url` the string representation, hence the `BacendInvitationAddr.to_url()` method for instance).

And of course the `to_http_redirection_url()` should use `urlsplit`/`urlunsplit` to cook the url (hence the complex corner cases should be taken care of by the stdlib instead of ourselves ^^)

Reply:
With something like that : 
`class BackendHTTPRedirectionAddr(BackendActionAddr):
    def __init__(
        self,
        organization_id: OrganizationID,
        invitation_type: InvitationType,
        token: UUID,
        **kwargs,
    ):
        if not hostname.startswith("parsec://"):
            hostname = "parsec://" + hostname
        super().__init__(**kwargs)
        self._organization_id = organization_id
        self._invitation_type = invitation_type
        self._token = token`
it should be alright to inherit from BackendAddr no ?
Because it will be more convenient for me to create this class for my PR https://github.com/Scille/parsec-cloud/pull/1294

Reply:
I'm actually thinking about creating the `BackendHTTPRedirectionAddr`, with inheritance from `BackendInvitationAddr` or from `BackendActionAddr` (in the second case, I will duplicate a big part of BackendinvitationAddr, but both part will be independant in the future...). I don't think that BackendAddr considering `parsec://` scheme is a problem : cf `BackendInvitationAddr.to_http_redirection_url()`

Reply:
Nevermind, I have no use of the class in the other PR finally ^^